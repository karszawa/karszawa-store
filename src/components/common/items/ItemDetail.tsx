import React from "react";
import { getImageUrl } from "~/lib/graphcool";
import styled from "styled-components";
import { A } from "../Anchor";
import Link from "next/link";
import { fakeText, LoadingSpan } from "~/styles/loading";

interface ItemDetailProps {
  id: string;
  name: string;
  description: string;
  file: { secret: string };
}

const ItemDetail: React.FC<ItemDetailProps> = ({
  id,
  name,
  description,
  file
}: ItemDetailProps) => {
  const imageUrl = getImageUrl({
    secret: file.secret,
    filename: name,
    size: "500x500"
  });

  return (
    <Container>
      <H1>{name}</H1>
      <Img src={imageUrl} alt={name} />
      <Link href={`/transactions/buy?itemId=${id}`}>
        <BuyA>Buy!</BuyA>
      </Link>
      <Description>{description}</Description>
    </Container>
  );
};

export default ItemDetail;

const Container = styled.article`
  align-self: center;
  max-width: 500px;
`;

const H1 = styled.h1`
  text-align: center;
`;

const Img = styled.img`
  width: 100%;
  border: 2px solid #eee;
  border-radius: 8px;
`;

const BuyA = styled(A)`
  justify-content: center;
  font-size: 32px;
  border: 1px solid #333;
  border-radius: 4px;
  margin-top: 16px;

  &:hover {
    color: white;
    background-color: #333;
    font-weight: bold;
    opacity: 1;
  }
`;

const Description = styled.p``;

export const ItemDetailLoading: React.FC<{}> = () => (
  <ContainerLoading>
    <H1>
      <LoadingSpan>Loading title ...</LoadingSpan>
    </H1>
    <ImgLoading />
    <Description>
      <LoadingSpan>{fakeText}</LoadingSpan>
    </Description>
  </ContainerLoading>
);

const ContainerLoading = styled.div`
  width: 100%;
  max-width: 400px;
  align-self: center;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ImgLoading = styled.div`
  width: 100%;
  height: 400px;
  border-radius: 8px;
  background-color: #eee;
  margin-bottom: 32px;
`;
