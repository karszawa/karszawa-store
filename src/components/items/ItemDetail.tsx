import React from "react";
import { GetItemData } from "~/queries/items";
import { getImageUrl } from "~/lib/graphcool";
import styled from "styled-components";
import { A } from "../common/Anchor";
import Link from "next/link";
import { fakeText, LoadingSpan } from "~/styles/loading";

interface ItemDetailProps {
  data: GetItemData;
}

const ItemDetail: React.FC<ItemDetailProps> = ({ data }: ItemDetailProps) => {
  const item = data.Item;
  const imageUrl = getImageUrl({
    secret: item.file.secret,
    filename: item.name,
    size: "500x500"
  });

  return (
    <Container>
      <H1>{item.name}</H1>
      <Img src={imageUrl} alt={item.name} />
      <Link href={`/transaction/buy?itemId=${item.id}`}>
        <BuyA>Buy!</BuyA>
      </Link>
      <Description>{item.description}</Description>
    </Container>
  );
};

export default ItemDetail;

const Container = styled.article`
  align-self: center;
  max-width: 500px;
`;

const H1 = styled.h1`
  text-align: center;
`;

const Img = styled.img`
  width: 100%;
  border: 2px solid #eee;
  border-radius: 8px;
`;

const BuyA = styled(A)`
  justify-content: center;
  font-size: 32px;
  border: 1px solid #333;
  border-radius: 4px;
  margin-top: 16px;

  &:hover {
    color: white;
    background-color: #333;
    font-weight: bold;
    opacity: 1;
  }
`;

const Description = styled.p``;

export const ItemDetailLoading: React.FC<{}> = () => (
  <ContainerLoading>
    <H1>
      <LoadingSpan>Loading title ...</LoadingSpan>
    </H1>
    <ImgLoading />
    <Description>
      <LoadingSpan>{fakeText}</LoadingSpan>
    </Description>
  </ContainerLoading>
);

const ContainerLoading = styled.div`
  width: 100%;
  max-width: 400px;
  align-self: center;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ImgLoading = styled.div`
  width: 100%;
  height: 400px;
  border-radius: 8px;
  background-color: #eee;
  margin-bottom: 32px;
`;
