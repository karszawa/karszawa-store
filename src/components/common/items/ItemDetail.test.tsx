import React from "react";
import { render } from "@testing-library/react";
import "jest-dom/extend-expect";
import ItemDetail from "./ItemDetail";

test("Render ItemDetail", () => {
  const { asFragment } = render(
    <ItemDetail
      id="123"
      name="card"
      description="awesome product"
      file={{ secret: "foo" }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
