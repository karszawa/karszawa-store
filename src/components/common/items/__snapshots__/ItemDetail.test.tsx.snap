// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ItemDetail 1`] = `
<DocumentFragment>
  <article
    class="ItemDetail__Container-sc-1ulk2l1-0 EToxH"
  >
    <h1
      class="ItemDetail__H1-sc-1ulk2l1-1 khVAR"
    >
      card
    </h1>
    <img
      alt="card"
      class="ItemDetail__Img-sc-1ulk2l1-2 xKYnk"
      src="https://images.graph.cool/v1/cjxbffaco1vxd0193t5giyfzl/foo/500x500/card"
    />
    <a
      class="Anchor__A-xh1dxk-0 ItemDetail__BuyA-sc-1ulk2l1-3 kFZtWn"
    >
      Buy!
    </a>
    <p
      class="ItemDetail__Description-sc-1ulk2l1-4 haRufg"
    >
      awesome product
    </p>
  </article>
</DocumentFragment>
`;
